### PR TITLE
test: add e2e tests for prompts functionality

### LINKED ISSUE #233: write e2e tests for prompts functionality
once prompts are merged #229 , we need to write some e2e tests for the functionality.

### PR DESCRIPTION
**Pull request by optimus-fulcria**
# Description

Add comprehensive e2e tests for the prompts functionality, covering the full prompt lifecycle and edge cases.

Closes #233

**New test cases added:**
- `TestPromptMultipleVersions` - publish v1 and v2, verify both via API, check latest resolution, list all versions, delete v1 and confirm v2 survives
- `TestPromptContentIntegrity` - round-trip publish/retrieve verifying exact content and description match
- `TestPromptShowNonExistent` - show a prompt that doesn't exist
- `TestPromptDeleteNonExistent` - delete a prompt that doesn't exist (expects failure)
- `TestPromptAPINotFound` - direct API 404 checks for nonexistent prompts and nonexistent versions of existing prompts
- `TestPromptPublishYAMLWithFlagOverrides` - verify CLI flags (--name, --description) override YAML values
- `TestPromptDryRunDoesNotCreate` - verify --dry-run does not persist the prompt (API returns 404 after)
- `TestPromptListOutputJSON` - verify --output json produces valid JSON
- `TestPromptDeleteThenShowReturnsNotFound` - publish, delete, then verify 404 via API
- `TestPromptPublishValidation/directory_instead_of_file` - pass a directory path instead of file

**Existing tests preserved unchanged:**
- `TestPromptPublishListShowDelete` (full lifecycle)
- `TestPromptPublishYAML`
- `TestPromptPublishDryRun`
- `TestPromptPublishValidation` (missing_name, missing_version, nonexistent_file)
- `TestPromptDeleteValidation` (missing_version_flag)

# Change Type

/kind feature

# Changelog

```release-note
Add comprehensive e2e tests for prompts: multi-version workflows, API verification, content integrity, error cases, and edge cases.
```

# Additional Notes

All new tests follow existing e2e patterns (build tag, helper functions, unique names, cleanup). Verified compilation with `go vet -tags e2e ./e2e/...`.

### CODE REVIEW

File: e2e/prompt_test.go [hunk 1]
```diff
@@ -212,3 +227,602 @@ func TestPromptDeleteValidation(t *testing.T) {
...
+
+	// Show (without version) should return latest
+	t.Run("show_returns_latest", func(t *testing.T) {
+		result := RunArctl(t, tmpDir,
+			"prompt", "show", promptName,
+			"--output", "json",
+			"--registry-url", regURL,
+		)
+		RequireSuccess(t, result)
+		RequireOutputContains(t, result, promptName)
```
Comment (inFocus7) [inline e2e/prompt_test.go:286]: we should check the latest (v2) was returned with `show`
```suggestion
		RequireOutputContains(t, result, promptName)
		RequireOutputContains(t, result, v2)
```

File: e2e/prompt_test.go [hunk 2]
```diff
@@ -212,3 +227,602 @@ func TestPromptDeleteValidation(t *testing.T) {
...
+		if promptResp.Prompt.Name != promptName {
+			t.Errorf("name = %q, want %q", promptResp.Prompt.Name, promptName)
+		}
+		if promptResp.Prompt.Version != v2 {
+			t.Errorf("version = %q, want %q", promptResp.Prompt.Version, v2)
+		}
+		if promptResp.Prompt.Content != v2Content {
+			t.Errorf("content = %q, want %q", promptResp.Prompt.Content, v2Content)
+		}
+	})
```
Comment (inFocus7) [inline e2e/prompt_test.go]: nit: it looks like we could dedupe a chunk of this, i think we could do something like below (although I heavily simplified the logic)

```go
versionTestCases := struct{
  version string
  expectedPrompt *Prompt
}{ {v1, expected}, {v2, expected}}

for test {
   url := regURL + "/prompts/" + promptName + "/versions/" +  test.version
   ...
   if response != test.expectedPrompt
} 
```

not a blocker though, just a thought on deduping 👍🏼

File: e2e/prompt_test.go [hunk 3]
```diff
@@ -212,3 +227,602 @@ func TestPromptDeleteValidation(t *testing.T) {
...
+				Count int `json:"count"`
+			} `json:"metadata"`
+		}
+		if err := json.Unmarshal(body, &versionsResp); err != nil {
+			t.Fatalf("failed to parse versions response: %v", err)
+		}
+
+		if versionsResp.Metadata.Count < 2 {
+			t.Errorf("expected at least 2 versions, got count=%d", versionsResp.Metadata.Count)
+		}
```
Comment (inFocus7) [inline e2e/prompt_test.go]:
```suggestion
		if versionsResp.Metadata.Count != 2 {
			t.Errorf("expected 2 versions, got count=%d", versionsResp.Metadata.Count)
		}
```

nit: just being more explicit to avoid possible unknown bug (e.g. accidentally duplicating prompts)

**Review by inFocus7 — COMMENTED** [review]
minor nits, but overall lgtm! the ones 100% worth adding imo are the `RequireOutputContains(t, result, v2)` & `versionsResp.Metadata.Count != 2` checks, the other request would just be nice for deduplication efforts.
